src/socket-io.service.ts

```diff
@@ -36,8 +36,11 @@ export class WrappedSocket {
     this.ioSocket.once(eventName, callback);
   }
 
-  connect() {
-    return this.ioSocket.connect();
+  connect(callback: (err: any) => void | undefined) {
+    if (callback === undefined) {
+      return this.ioSocket.connect();
+    }
+    return this.ioSocket.connect(callback);
```

Comment:
```suggestion
  connect(callback?: (err: any) => void) {
    return this.ioSocket.connect(callback);
```

You can use the `?` modifier for optional arguments. Also, calling a function with an 'undefined` argument should be the same as calling it without that argument.